m2cgen/ast.py

```diff
@@ -230,3 +230,34 @@ def __init__(self, expr, to_reuse=False):
     def __str__(self):
         args = ",".join([str(self.expr), "to_reuse=" + str(self.to_reuse)])
         return "SubroutineExpr(" + args + ")"
+
+
+NESTED_EXPRS_MAPPINGS = [
+    ((BinExpr, CompExpr), lambda e: [e.left, e.right]),
+    (PowExpr, lambda e: [e.base_expr, e.exp_expr]),
+    (VectorVal, lambda e: e.exprs),
+    (IfExpr, lambda e: [e.test, e.body, e.orelse]),
```

Comment:
I think these expr can be wrapped into a tuples too for the consistent interface. Like, `((PowExpr), lambda e: [e.base_expr, e.exp_expr]),`

Reply:
Sorry, I'm not 100% sold on this :D